provider/boltmem/boltmem.go

```diff
@@ -261,15 +261,15 @@ func (s *Silences) Mutes(lset model.LabelSet) bool {
 
 // All returns all existing silences.
 func (s *Silences) All() (*types.SilencesQueryResponse, error) {
-	return s.Query(len(s.k), 0)
+	return s.Query(len(s.sils), 0, types.ByCreatedAt)
 }
 
 // Query returns n silences starting at page offset o.
-func (s *Silences) Query(n, o int) (*types.SilencesQueryResponse, error) {
+func (s *Silences) Query(n, o int, fn types.SilencesLessFunc) (*types.SilencesQueryResponse, error) {
```

Comment:
Yeah, so let's leverage the type system to guarantee that.

On Wed, Aug 24, 2016, 20:57 stuart nelson <notifications@github.com> wrote:

> In provider/boltmem/boltmem.go
> <https://github.com/prometheus/alertmanager/pull/475#discussion_r76117385>
> :
>
> >  }
> >
> >  // Query returns n silences starting at page offset o.
> > -func (s *Silences) Query(n, o int) (*types.SilencesQueryResponse, error) {
> > +func (s *Silences) Query(n, o int, fn types.SilencesLessFunc) (*types.SilencesQueryResponse, error) {
>
> I understand the second point regarding descriptive names, but why would
> uint be better? Is it because we'll never need a negative value?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/prometheus/alertmanager/pull/475/files/3fffc9f80e61e7c51765b0b5ec65c0633910cdd5#r76117385>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAANaJ0NeDxOqOqhAc_i7JOOkqr86D0Wks5qjJQTgaJpZM4JsKty>
> .
>